app/controllers/api/v1/sessions_controller.rb

```diff
@@ -0,0 +1,18 @@
+class Api::V1::SessionsController < ApplicationController
+  def create
+    user = User.find_by(email: session_params[:email])
+    if user.present? && user.authenticate(session_params[:password])
+      render json: UserSerializer.new(user)
+    elsif user.nil?
+      render json: { error: 'Invalid Credentials', status: 404 }, status: 404
+    else
+      render json: { error: 'Invalid password', status: 401 }, status: 401
```

Comment:
Is this cool? Is this where this should happen? Error PORO? Error Serializer? 

![](https://media0.giphy.com/media/d1E1YlkOTe4IfdNC/giphy.gif?cid=5a38a5a2e86ep2r41qbwfribtjup2yzejh5vddes5zz05hh0&rid=giphy.gif)